packages/core/src/codewhisperer/util/customizationUtil.ts

```diff
@@ -116,25 +112,27 @@ export const getSelectedCustomization = (): Customization => {
     if (selectedCustomization && selectedCustomization.name !== '') {
         return selectedCustomization
     } else {
-        const customizationFeature = FeatureConfigProvider.getFeature(Features.customizationArnOverride)
-        const arnOverride = customizationFeature?.value.stringValue
-        const customizationOverrideName = customizationFeature?.variation
-        if (arnOverride === undefined) {
-            return baseCustomization
-        } else {
-            return {
-                arn: arnOverride,
-                name: customizationOverrideName,
-                description: baseCustomization.description,
-            }
-        }
+        return baseCustomization
     }
 }
 
-export const setSelectedCustomization = async (customization: Customization) => {
+/**
+ * @param customization customization to select
+ * @param isOverride if the API call is made from us (Q) but not users' intent, set isOverride to TRUE
+ * Override happens when ALL following conditions are met
+ *  1. service returns non-empty override customization arn, refer to [featureConfig.ts]
+ *  2. the override customization arn is different from the previous override customization if any. The purpose is to only do override once on users' behalf.
+ */
+export const setSelectedCustomization = async (customization: Customization, isOverride: boolean = false) => {
     if (!AuthUtil.instance.isValidEnterpriseSsoInUse() || !AuthUtil.instance.conn) {
         return
     }
+    if (isOverride) {
```

Comment:
Just for my own understanding -- this means that the server has received a non empty override customization arn and we want to check whether or not its a new arn. If the same arn is found then we're good to go since there's nothing to do. Otherwise override the customization?

Is that correct?

Reply:
> his means that the server has received a non empty override customization arn and we want to check whether or not its a new arn.

this is correct.


> If the same arn is found then we're good to go since there's nothing to do. Otherwise override the customization?

yes and no. If the customization arn is same as the last one (which we had done override once), we do nothing because it's intrusive if we change users' selection every time when there is a non-null override customization arn.